finetune/target_models/deployment_model.py

```diff
@@ -0,0 +1,344 @@
+import tensorflow as tf
+import numpy as np
+import tqdm
+import math
+import itertools
+import logging
+import pandas as pd
+from imblearn.over_sampling import RandomOverSampler
+from sklearn.utils import shuffle
+from collections import namedtuple
+from tensorflow.python.data import Dataset
+
+from finetune.config import get_config
+from finetune.saver import Saver, InitializeHook
+from finetune.base import BaseModel
+from finetune.target_models.comparison import ComparisonPipeline
+from finetune.target_models.sequence_labeling import SequencePipeline, SequenceLabeler
+from finetune.target_models.association import AssociationPipeline
+from finetune.target_models.classifier import ClassificationPipeline
+from finetune.base_models import GPTModel, GPTModelSmall, BERTModelCased, GPT2Model
+from finetune.model import get_model_fn, get_separate_model_fns, PredictMode
+from finetune.encoding.target_encoders import OneHotLabelEncoder
+from finetune.errors import FinetuneError
+from finetune.input_pipeline import BasePipeline
+
+LOGGER = logging.getLogger('finetune')
+
+class TaskMode:
+    SEQUENCE_LABELING = "Sequence_Labeling"
+    CLASSIFICATION = "Classification"
+    COMPARISON = "Comparison"
+    ASSOCIATION = "Association"
+
+
+class DeploymentPipeline(BasePipeline):
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.pipeline_type = None
+        self.pipeline = None
+
+    def _target_encoder(self):
+        raise NotImplementedError
+
+    def _dataset_without_targets(self, Xs, train):
+        if not callable(Xs):
+            Xs_fn = lambda: self.wrap_tqdm(Xs, train)
+        else:
+            Xs_fn = lambda: self.wrap_tqdm(Xs(), train)
+        
+        dataset_encoded = lambda: itertools.chain.from_iterable(map(self.get_text_token_mask, Xs_fn()))
+        types, shapes = self.feed_shape_type_def()
+        return Dataset.from_generator(dataset_encoded, types[0])
+        
+    def pipe_gen(self):
```

Comment:
I think that this is only ever called with next meaning it doesn't need to be a generator it can just return?

Reply:
I think especially as it isnt a generator anymore it can get a better name. maybe `get_active_pipeline`